packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -982,8 +982,13 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
       // Subscribe to all changes by default
       observedBits = MAX_SIGNED_31_BIT_INT;
     }
+
     // Store the observedBits on the fiber's stateNode for quick access.
-    workInProgress.stateNode = observedBits;
+    if (oldProps === null) {
```

Comment:
Props nullability have had other meanings in the past and doesn't have a defined meaning now but might get one. 

What is this meant to test? If it is to test that this is first mount, it should check for `current === null`, but since this is really just testing whether `stateNode` is null you might as well just make that explicit (Flow will want you to anyway once we properly type stateNode).